### PR TITLE
{phys}[GCCcore/12.3.0] PyAEDT v0.8.7

### PR DESCRIPTION
**Pull request by appolloford**
(created using `eb --new-pr`)

- [x] AEDT (https://github.com/easybuilders/easybuild-easyconfigs/pull/20309) as its backend

### CODE REVIEW

**Comment by appolloford** [comment]
Test report by @appolloford
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/appolloford/b0107145976ed54f2d50986f534c7b29 for a full test report.

**Comment by jfgrimm** [comment]
> AssertionError: Should not have multiple variants of dependencies.
Found 2 variants of 'dotNET-Core' dependency in easyconfigs using 'GCCcore-12.3' toolchain generation
* version: 8.0.203; versionsuffix:  as dep for {'PyAEDT-0.8.7-GCCcore-12.3.0.eb'}
  * version: 8.0; versionsuffix:  as dep for {'PyAEDT-0.8.7-GCCcore-12.3.0.eb'}

**Comment by Micket** [comment]
I asked @appolloford to also update these tests to cover the net dotNET-Core, which should be treated identically to the Java wrappers (since they are).

**Comment by jfgrimm** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/p/PyAEDT/PyAEDT-0.8.7-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+builddependencies = [
+    ('binutils', '2.40'),
+]
+
+dependencies = [
+    ('dotNET-Core', '8.0', '', SYSTEM),
+    ('Python', '3.11.3'),
+    ('Python-bundle-PyPI', '2023.06'),
+    ('Pillow', '10.0.0'),
+    ('hatchling', '1.18.0'),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/p/PyAEDT/PyAEDT-0.8.7-GCCcore-12.3.0.eb:22]: This should be a build dependency

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/p/PyAEDT/PyAEDT-0.8.7-GCCcore-12.3.0.eb (hunk 1 above)
Comment (appolloford) [reply easybuild/easyconfigs/p/PyAEDT/PyAEDT-0.8.7-GCCcore-12.3.0.eb]: Thanks! Fixed!

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/c04b7bd46947d66e86508480b9b5ec35 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Pull request merged**